client/my-sites/people/edit-team-member-form/with-update-user.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useDispatch } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import useUpdateUserMutation from 'calypso/data/users/use-update-user-mutation';
+import { errorNotice, successNotice, plainNotice } from 'calypso/state/notices/actions';
+import { useTranslate } from 'i18n-calypso';
+
+const useSuccessNotice = ( isSuccess, user ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		isSuccess &&
+			dispatch(
+				successNotice(
+					translate( 'Successfully updated @%(user)s', {
+						args: { user: user?.login },
+						context: 'Success message after a user has been modified.',
+					} ),
+					{ id: 'update-user-notice' }
+				)
+			);
+		// Note: We only want to run this effect in case `isSuccess`
+		// changes and ignore changes to other deps like `user`.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@flootr I finally figured out the "right" way how to write the `useSuccessNotice` hook 🙂 Here it is:
```js
const useSuccessNotice = ( isSuccess, user ) => {
  const showNotice = React.useRef();
  const dispatch = useDispatch();
  const translate = useTranslate();

  // update the `showNotice` callback when its dependencies change
  React.useEffect( () => {
    showNotice.current = () =>
      dispatch(
        successNotice(
          translate( 'Successfully updated @%(user)s', { args: { user: user?.login } } )
        )
      );
  }, [ dispatch, translate, user ] );

  // call the `showNotice` callback when its conditions are triggered
  React.useEffect( () => {
    isSuccess && showNotice.current();
  }, [ isSuccess ] );
};
```

The trick is to use ref and two effects. The first effect runs when the "show notice" callback's dependencies change (`dispatch`, `translate` or `user`) and stores the up-to-date callback to the ref. The second effect then calls the stored callback as triggered by the `isSuccess` dependency.

100% of the inspiration comes from this blog post by Dan Abramov about a `useInterval` hook:

https://overreacted.io/making-setinterval-declarative-with-react-hooks/

They use exactly the same technique there. One effect updates the interval callback without clearing and re-creating the `setInterval` loop. A second effect resets the `setInterval` when its `delay` prop changes.

Reply:
Awesome, thanks @jsnajdr for the additional input. I'll update this PR to use this pattern. For now this PR has to wait for another PR anyway because it currently breaks some e2e tests related to deleting users.